Mathlib/Analysis/CStarAlgebra/CStarMatrix.lean

```diff
@@ -0,0 +1,694 @@
+/-
+Copyright (c) 2024 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import Mathlib.Analysis.CStarAlgebra.Module.Constructions
+import Mathlib.Analysis.Matrix
+import Mathlib.Topology.UniformSpace.Matrix
+
+/-!
+# Matrices with entries in a C⋆-algebra
+
+This file creates a type copy of `Matrix m n A` called `CStarMatrix m n A` meant for vectors with
+entries in a C⋆-algebra `A`. Its action on `WithCStarModule (n → A)` (via `Matrix.mulVec`) gives
+it the operator norm, and this norm makes `CStarMatrix n n A` a C⋆-algebra.
+
+## Main declarations
+
++ `CStarMatrix m n A`: the type copy
++ `CStarMatrix.instCStarRing`: square matrices with entries in a C⋆-algebra form a C⋆-algebra
+
+## Implementation notes
+
+The norm on this type induces the product uniformity and bornology, but these are not defeq to
+`Pi.uniformSpace` and `Pi.instBornology`. Hence, we prove the equality to the Pi instances and
+replace the uniformity and bornology by the Pi ones when registering the
+`NormedAddCommGroup (CStarMatrix m n A)` instance. See the docstring of the `TopologyAux` section
+below for more details.
+-/
+
+set_option maxSynthPendingDepth 2
```

Comment:
Where exactly is this needed? Can you just activate it locally (even in the proof) so we can see exactly where the problems occur?